doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,255 @@
+.. _neural_network:
+
+==================================
+Neural network models (supervised)
+==================================
+
+.. currentmodule:: sklearn.neural_network
+
+
+.. _elm:
+
+Extreme Learning Machines
+=========================
+
+**Extreme Learning Machine (ELM)** is a supervised nonlinear learning 
+algorithm that takes the following consecutive steps for training:
+
+  *  it applies a random projection to the input space, onto a possibly higher 
+     dimensional space;
+  *  the result passes through an element-wise non-linear activation function,
+     typically a Rectified Linear Unit or a sigmoid function (tanh or logistic); and
+  *  last, it trains a linear one vs. rest classifier or a multi-output ridge 
```

Comment:
I would say "it fits a supervised linear ridge-model, that is a least square model with linear penalty, for both regression and classification". At least that seems to be what is implemented. Your current formulation leaves it open what loss  and penalty are for the classification case.